Runtime/src/index.ts

```diff
@@ -328,6 +392,51 @@ export class SwiftRuntime {
             return result;
         };
 
+        const syncAwait = (
+            promise: Promise<any>,
+            kind_ptr?: pointer,
+            payload1_ptr?: pointer,
+            payload2_ptr?: pointer
+        ) => {
+            if (!this.instance || !this.instanceIsAsyncified) {
+                throw new Error("Calling async methods requires preprocessing Wasm module with `--asyncify`");
+            }
+            const exports = (this.instance
+                .exports as any) as AsyncifyExportedFunctions;
+            if (!this.isSleeping) {
+                // Fill in the data structure. The first value has the stack location,
+                // which for simplicity we can start right after the data structure itself.
+                const int32Memory = new Int32Array(memory().buffer);
+                const ASYNCIFY_STACK_POINTER = 16; // Where the unwind/rewind data structure will live.
```

Comment:
Is it safe to use static memory space in linear memory without malloc?

Reply:
Great question which I do not know the answer to :) This is copied from the original implementation, but let me dig deeper in Emscripten's production code for this.